### PR TITLE
Add Owner Reference to VM Snapshots

### PR DESCRIPTION
**Pull request by glekner**

### CODE REVIEW

**Comment by glekner** [comment]
@suomiy can you lgtm

File: frontend/packages/kubevirt-plugin/src/k8s/wrapper/common/k8s-resource-wrapper.ts [hunk 1]
```diff
@@ -79,4 +80,22 @@ export abstract class K8sResourceWrapper<
     }
     return (this as any) as SELF;
   };
+
+  addOwnerReferences = (...additionalOwnerReferences) => {
+    if (!getOwnerReferences(this.data)) {
```
Comment (suomiy) [inline frontend/packages/kubevirt-plugin/src/k8s/wrapper/common/k8s-resource-wrapper.ts:85]: you can use `ensurePath` function instead

File: frontend/packages/kubevirt-plugin/src/k8s/wrapper/common/k8s-resource-wrapper.ts [hunk 2]
```diff
@@ -79,4 +80,22 @@ export abstract class K8sResourceWrapper<
     }
     return (this as any) as SELF;
   };
+
+  addOwnerReferences = (...additionalOwnerReferences) => {
```
Comment (suomiy) [inline frontend/packages/kubevirt-plugin/src/k8s/wrapper/common/k8s-resource-wrapper.ts:84]: can you please add a type?

File: frontend/packages/kubevirt-plugin/src/k8s/wrapper/common/k8s-resource-wrapper.ts [hunk 3]
```diff
@@ -79,4 +80,22 @@ export abstract class K8sResourceWrapper<
     }
     return (this as any) as SELF;
   };
+
```
Comment (suomiy) [inline frontend/packages/kubevirt-plugin/src/k8s/wrapper/common/k8s-resource-wrapper.ts:83]: we can add this function to K8sResourceObjectWithTypePropertyWrapper as well for consistency

**Review by suomiy — COMMENTED** [review]
Are you planning to add the message to the VM delete modal in another PR?

File: frontend/packages/kubevirt-plugin/src/k8s/wrapper/common/k8s-resource-wrapper.ts (hunk 1 above)
Comment (glekner) [reply frontend/packages/kubevirt-plugin/src/k8s/wrapper/common/k8s-resource-wrapper.ts:85]: `ensurePath` doesn't initialize `ownerReferences` as an array. `getOwnerReferences(this.data)` returns `{}` if there are no owners, resulting in errors inside the array functions

**Comment by glekner** [comment]
@suomiy applied.
> Are you planning to add the message to the VM delete modal in another PR?
this should go through designers yes

File: frontend/packages/kubevirt-plugin/src/k8s/wrapper/common/k8s-resource-wrapper.ts (hunk 1 above)
Comment (suomiy) [reply frontend/packages/kubevirt-plugin/src/k8s/wrapper/common/k8s-resource-wrapper.ts:85]: please look at the implementation, it is possible to pass the desired structure

**Comment by suomiy** [comment]
> > Are you planning to add the message to the VM delete modal in another PR?
> 
> this should go through designers yes

I think we already got response from designers in the snapshot thread. Let's just add a simple message to the delete vm modal. And we can ping designers in that PR and ask them what they think.

**Comment by glekner** [comment]
image

Added message to delete vm modal
@suomiy @matthewcarleton

File: frontend/packages/kubevirt-plugin/src/k8s/wrapper/common/k8s-resource-wrapper.ts (hunk 1 above)
Comment (glekner) [reply frontend/packages/kubevirt-plugin/src/k8s/wrapper/common/k8s-resource-wrapper.ts:85]: updated

**Comment by matthewcarleton** [comment]
Looks good but can you adjust the text to be
**Warning**: All snapshots of this virtual machine will be deleted as well.

**Review by matthewcarleton — APPROVED** [review]
LGTM!

**Pull request merged**